audioplayer.js

```diff
@@ -1,135 +1,208 @@
                 watcher.on("delete", function(e) {
-                    watcher.unwatch(e.path);
-                    if (audioDoc.tab) {
-                        showAlert(
-                            "File no longer available", 
-                            e.path + " was removed or no longer available.",
-                            null,
-                            function() {
-                                audioDoc.tab.close();    
-                            }
-                        );
-                    }
+                    var path = e.path;
+                    var tab = watchedPaths[path];
+                    // ensure path is being watched
+                    if (_.isUndefined(tab))
+                        return;
+                    unwatch(path);
+                    // alert user and close tab
+                    showAlert(
+                        "File is no longer available",
+                        path + " is no longer available",
+                        null,
+                        tab.close
+                    );
                 });
             });
-            
-            // handle when tab becomes active (e.g., after refresh)
+
+            // handle when tab for audio file becomes active
             plugin.on("documentActivate", function(e) {
                 var audioDoc = e.doc;
+                var session = audioDoc.getSession();
+                // hide current player from tab (if any)
+                if (currentSession && currentSession !== session) {
+                    currentSession.audio.style.display = "none";
+                }
+
+                // update current session
+                currentSession = session;
+
+                // ensure new player is attached to container
+                if (!container.contains(currentSession.audio)) {
+                    container.appendChild(currentSession.audio);
+                }
+                // ensure new player is visible
+                currentSession.audio.style.display = "initial";
+
+                // set/update player src URL
+                setPath(audioDoc);
+
+                // preserve playing or pausing state (e.g., when moving player to another pane)
+                if (currentSession.paused === false && currentSession.audio.paused === true) {
+                    currentSession.audio.play();
+                }
+            });
+
+            // handle document unloading (e.g., when tab is closed or moved to another pane)
+            plugin.on("documentUnload", function(e) {
+                var audioDoc = e.doc;
+                var audio = audioDoc.getSession().audio;
+                // remove player from pane
+                container.removeChild(audio);
+
                 var path = audioDoc.tab.path;
-                setPath(path, audioDoc);
+                // alert user if file was deleted before unwatching
+                watcher.check(path);
+                setTimeout(function() {
```

Comment:
Hmmmm. Do we want to override the default behavior here? Maybe I'm misreading the code, but if the user deletes the file in the GUI they will get two dialogs: a file deletion confirmation and then a file went away dialog, the latter from this plugin? I would think only the first dialog is necessary since that's clear enough about what's happened. If we want to override the behavior we should probably do it at a lower level so that it applies to all editors?